### DISCUSSION TITLE
BasicVSR++2 and Training Speed

### DISCUSSION THREAD

**Discussion by mikolez** [open-mmlab/mmediting#1403] [How-to] [answered]
### Model/Dataset/Scheduler description

In the technical report for deblurring and denoising (https://arxiv.org/pdf/2204.05308.pdf), there are two versions of the BasicVSR++ for video denoising, the one which reduces the resolution by 2 (BasicVSR++2, superior performance but less speed) and by 4 (BasicVSR++4, more speed but worse performance). Current code, as I understand, implements the version which reduces the resolution by 4. How do I change the code exactly to switch to the BasicVSR++2 (the one which reduces the resolution by 2)? Or could you please share the code?

In addition, I am trying to speed up the training for denoising. I tried increasing samples_per_gpu config variable from 1 to 3 and decreasing the number of iterations from 600,000 to 200,000 (so that the amount of data that the model uses for training is the same), which helped to improve the time quite well (from 13 days to 6 days of training). Is there any other way to make the training faster without hurting the performance after the training?

Thank you!

### Open source status

- [ ] The model implementation is available
- [ ] The model weights are available.

**Comment by plyfager** [comment]
Hi @mikolez , since I'm not focused at this area, I suggest you make a [discussion](https://github.com/open-mmlab/mmediting/discussions) for help. Or maybe @ckkelvinchan would help.

**Comment by plyfager** [comment]
I'll try to find the answers and reply later.

**Comment by ckkelvinchan** [comment, marked as answer]
Hi @mikolez, you can simply reduce the number of downsampling (i.e., strided conv) and upsampling (i.e., pixel shuffle) from 2 to 1. I will update the code later, but I am currently busy with some other stuff.

About the training speed, this [paper](https://arxiv.org/abs/2205.05069) may help, but it requires dynamic sequence length and batch size. It may require additional code modifications.
